fetch/api/basic/request-upload.js

```diff
@@ -18,6 +18,20 @@ function testUpload(desc, url, method, body, expectedBody) {
   }, desc);
 }
 
+function testUploadFailure(desc, url, method, body) {
+  const requestInit = {"method": method};
+  promise_test(test => {
+    if (typeof body === "function")
+      body = body();
+    if (body)
+      requestInit["body"] = body;
+    return fetch(url, requestInit).then(() => {
```

Comment:
This should use `return promise_rejects(new TypeError(), fetch(url, requestInit))`